lib/matplotlib/axes/_axes.py

```diff
@@ -2581,6 +2582,123 @@ def barh(self, y, width, height=0.8, left=None, *, align="center",
                            align=align, **kwargs)
         return patches
 
+    def bar_label(self, container, labels=None, *, fmt="%g", label_type="edge",
```

Comment:
do we want to take new style format strings instead?  See arguments both ways.

Reply:
do we? both ok for me, currently using old style just for consistency with `clabel`.

Reply:
One *could* also support both using
~~~
try:
    label = fmt % value
except TypeError:
    label = fmt.format(value)
~~~
But I'm not sure if that's worth it or if we even want that redundancy.

I'm fine with sticking to %-formatting, but don't have a strong opinion on that.

Reply:
Hi @tacaswell, could you make a final decision? new, old or both?

Reply:
I think we should stick to old-style formatting for consistency throughout the lib.